lib/fibonacci.rb

```diff
@@ -1,8 +1,13 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - o(n)
+# Space Complexity - o(n)  (should be O(n))
 # Hint, you may want a recursive helper method
 def fibonacci(n)
```

Comment:
This is the brute force O(2<sup>n</sup>) solution.  Check the notes and see if you can approach this a better way.